tests/ImageSharp.Tests/TestUtilities/FeatureTesting/FeatureTestRunner.cs

```diff
@@ -0,0 +1,231 @@
+            where T : IXunitSerializable
+            => BasicSerializer.Deserialize<T>(value);
+
+        // TODO: Write runner test and use this.
+        private static void AssertHwIntrinsicsFeatureDisabled(HwIntrinsics intrinsics)
+        {
+            switch (intrinsics)
+            {
+                case HwIntrinsics.DisableSIMD:
+                    Assert.False(Vector.IsHardwareAccelerated);
+                    break;
+#if SUPPORTS_RUNTIME_INTRINSICS
+                case HwIntrinsics.DisableHWIntrinsic:
+                    Assert.False(Vector.IsHardwareAccelerated);
+                    break;
+                case HwIntrinsics.DisableSSE:
+                    Assert.False(Sse.IsSupported);
+                    break;
+                case HwIntrinsics.DisableSSE2:
+                    Assert.False(Sse2.IsSupported);
+                    break;
+                case HwIntrinsics.DisableAES:
+                    Assert.False(Aes.IsSupported);
+                    break;
+                case HwIntrinsics.DisablePCLMULQDQ:
+                    Assert.False(Pclmulqdq.IsSupported);
+                    break;
+                case HwIntrinsics.DisableSSE3:
+                    Assert.False(Sse3.IsSupported);
+                    break;
+                case HwIntrinsics.DisableSSSE3:
+                    Assert.False(Ssse3.IsSupported);
+                    break;
+                case HwIntrinsics.DisableSSE41:
+                    Assert.False(Sse41.IsSupported);
+                    break;
+                case HwIntrinsics.DisableSSE42:
+                    Assert.False(Sse42.IsSupported);
+                    break;
+                case HwIntrinsics.DisablePOPCNT:
+                    Assert.False(Popcnt.IsSupported);
+                    break;
+                case HwIntrinsics.DisableAVX:
+                    Assert.False(Avx.IsSupported);
+                    break;
+                case HwIntrinsics.DisableFMA:
+                    Assert.False(Fma.IsSupported);
+                    break;
+                case HwIntrinsics.DisableAVX2:
+                    Assert.False(Avx2.IsSupported);
+                    break;
+                case HwIntrinsics.DisableBMI1:
+                    Assert.False(Bmi1.IsSupported);
+                    break;
+                case HwIntrinsics.DisableBMI2:
+                    Assert.False(Bmi2.IsSupported);
+                    break;
+                case HwIntrinsics.DisableLZCNT:
+                    Assert.False(Lzcnt.IsSupported);
+                    break;
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Runs the given test <paramref name="action"/> within an environment
+        /// where the given <paramref name="intrinsics"/> features.
+        /// </summary>
+        /// <param name="action">The test action to run.</param>
+        /// <param name="intrinsics">The intrinsics features.</param>
+        public static void RunWithHwIntrinsicsFeature(
```

Comment:
Anyways, great stuff! Would be nice to add tests for this, but this is currently blocked.

Reply:
I do have tests waiting to push. Just hit the executor issue so stopped. I’ll revert later, merge the current state and add as a separate PR.